app_dart/test/request_handlers/push_benchmark_to_center_test.dart

```diff
@@ -0,0 +1,97 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:cocoon_service/src/model/appengine/time_series.dart';
+import 'package:cocoon_service/src/model/appengine/time_series_value.dart';
+import 'package:metrics_center/flutter.dart' as mc;
+import 'package:test/test.dart';
+
+import '../src/datastore/fake_cocoon_config.dart';
+import '../src/request_handling/fake_logging.dart';
+
+void main() {
+  FakeConfig config;
+  FakeLogging logger;
+  PushBenchmarkToCenter handler;
+
+  setUp(() {
+    config = FakeConfig();
+    logger = FakeLogging();
+    handler = PushBenchmarkToCenter(config, logger: logger);
+  });
+
+  group('parseSinceMillis', () {
+    test('with missing parameter', () {
+      final int sinceMillis = handler.parseSinceMillis(<String, String>{});
+      expect(sinceMillis, isNull);
+    });
+
+    test('parses correctly', () {
+      final int sinceMillis =
+          handler.parseSinceMillis(<String, String>{'sinceMillis': '123'});
+      expect(sinceMillis, 123);
+    });
+  });
+
+  group('readTimeSeriesMap', () {
+    test('returns a map', () async {
+      final TimeSeries ts =
+          TimeSeries(key: config.db.emptyKey.append(TimeSeries, id: 'tsId'));
+      config.db.addValue(ts);
+
+      final Map<String, TimeSeries> map = await handler.readTimeSeriesMap();
+
+      expect(map, <String, TimeSeries>{'tsId': ts});
+    });
+  });
+
+  group('readTimeSeriesValue', () {
+    test('returns a list', () async {
+      final TimeSeries ts =
+          TimeSeries(key: config.db.emptyKey.append(TimeSeries, id: 'tsId'));
+      final TimeSeriesValue tsv = TimeSeriesValue(
+        key: ts.key.append(TimeSeriesValue, id: 1),
+        value: 2,
+        branch: 'master',
+      );
+      config.db.addValue(tsv);
+
+      final List<TimeSeriesValue> list = await handler.readTimeSeriesValue(100);
+
+      expect(list, <TimeSeriesValue>[tsv]);
+    });
+  });
+
+  group('transform', () {
+    test('returns a list', () async {
+      final TimeSeries ts =
+          TimeSeries(key: config.db.emptyKey.append(TimeSeries, id: 'tsId'));
+      final TimeSeriesValue tsv = TimeSeriesValue(
+        key: ts.key.append(TimeSeriesValue, id: 1),
+        value: 2,
+        branch: 'master',
+      );
+
+      final List<mc.MetricPoint> metricPoints = await handler
+          .transform(<String, TimeSeries>{'tsId': ts}, <TimeSeriesValue>[tsv]);
+
+      expect(metricPoints, <mc.MetricPoint>[
+        mc.MetricPoint(
+          2,
+          const <String, String>{
+            'gitRepo': 'flutter/flutter',
+            'gitRevision': null,
+            'branch': 'master',
+            'originId': 'devicelab',
+            'taskName': null,
+            'unit': null,
+            'name': null,
+            'archived': 'false',
+          },
+          'devicelab',
+        )
+      ]);
+    });
```

Comment:
Test for `destination` result is missing.

Reply:
Done.